src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -517,18 +517,19 @@ public function makePathRelative($endPath, $startPath)
      *  - existing files in the target directory will be overwritten, except if they are newer (see the `override` option)
      *  - files in the target directory that do not exist in the source directory will not be deleted (see the `delete` option)
      *
-     * @param string            $originDir The origin directory
-     * @param string            $targetDir The target directory
-     * @param \Traversable|null $iterator  Iterator that filters which files and directories to copy, if null a recursive iterator is created
-     * @param array             $options   An array of boolean options
-     *                                     Valid options are:
-     *                                     - $options['override'] If true, target files newer than origin files are overwritten (see copy(), defaults to false)
-     *                                     - $options['copy_on_windows'] Whether to copy files instead of links on Windows (see symlink(), defaults to false)
-     *                                     - $options['delete'] Whether to delete files that are not in the source directory (defaults to false)
+     * @param string            $originDir      The origin directory
+     * @param string            $targetDir      The target directory
+     * @param \Traversable|null $originIterator Iterator that filters which files and directories to copy, if null a recursive iterator is created
+     * @param array             $options        An array of boolean options
+     *                                          Valid options are:
+     *                                          - $options['override'] If true, target files newer than origin files are overwritten (see copy(), defaults to false)
+     *                                          - $options['copy_on_windows'] Whether to copy files instead of links on Windows (see symlink(), defaults to false)
+     *                                          - $options['delete'] Whether to delete files that are not in the source directory (defaults to false)
+     * @param \Traversable|null $targetIterator Iterator that filters which files and directory in target directory to delete if not in source directory (if $options['delete'] is true)
      *
      * @throws IOException When file type is unknown
      */
-    public function mirror($originDir, $targetDir, \Traversable $iterator = null, $options = [])
+    public function mirror($originDir, $targetDir, \Traversable $originIterator = null, $options = [], \Traversable $targetIterator = null)
```

Comment:
The `Filesystem` class is not final. We cannot add new parameters to the signature because that would break code that overrides this method.

Reply:
I don't see any way to solve this without an extra parameter (though we certainly don't have to rename the existing parameter). What do you propose as an alternative fix?

Reply:
You can leave the parameter commented out and use `func_num_args()` to check if it has been passed and `func_get_arg()` to fetch its value. We can truly add the parameter in Symfony 6 then and remove the `func_num_args()`/`func_get_arg()` quirk again.

But this is a pattern we usually apply in feature releases. We would basically introduce a deprecation this way and that should not happen in a bugfix release. Adding new parameters to public methods sounds like we're doing bit more than just a bugfix here.

Reply:
I agree it straddles bug fix and feature. Fundamentally, this is definitely a bug (that the iterator cannot be used as intended or documented), but there's no way to fix it (that I see) except to add a second parameter.

I'll attempt the `func_get_arg()` change as you describe, instead of adding the parameter to the function definition. I think this can still happen as a bug fix, the deprecation can happen as a follow-up issue. Let me know if you disagree.